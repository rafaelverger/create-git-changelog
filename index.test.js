const EventEmitter = require('events');
const fs = require('fs')

test('changelog creation', (done) => {
  jest.mock('child_process');
  const spawnMock = require('child_process').spawn;
  const commitHistory = fs.readFileSync('./__mock__/commit.history');

  const emitter = new EventEmitter();
  emitter.stderr = new EventEmitter();
  emitter.stdout = new EventEmitter();

  spawnMock.mockReturnValue(emitter);

  fs.watchFile('CHANGELOG.md', (curr, prev) => {
    const changelog = fs.readFileSync('CHANGELOG.md').toString();
    expect(changelog).toMatchSnapshot();
    done();
  });

  require('./index')();

  emitter.stdout.emit('data', commitHistory);
  emitter.emit('close', 0);
});
