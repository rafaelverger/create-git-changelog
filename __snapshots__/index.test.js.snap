// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changelog creation 1`] = `
"# CHANGELOG

## unpublished
- [FIX #17] Using new info to filter significant commits.
  and
  _Also related to:_ #18 #19
- Enhancing change infos while parsing commits.
  _Also related to:_ #7
- Add wercker badge to README
- [FIX #3] Using wercker as CI.
- [FIX #4] Build changes using commits msg and grouped by version.
- [FIX #1] Create method to parse commit text.

## 0.0.0-alpha
- Add package.json
- Initial commit
"
`;

exports[`changelog creation using more than 2 lines 1`] = `
"# CHANGELOG

## unpublished
- [FIX #17] Using new info to filter significant commits.
  and
  more than two lines 
  _Also related to:_ #18 #19
- Enhancing change infos while parsing commits.
  _Also related to:_ #7
- Add wercker badge to README
- [FIX #3] Using wercker as CI.
- [FIX #4] Build changes using commits msg and grouped by version.
- [FIX #1] Create method to parse commit text.

## 0.0.0-alpha
- Add package.json
- Initial commit
"
`;

exports[`changelog creation with only unpublished changes 1`] = `
"# CHANGELOG

## unpublished
- [FIX #7] Using new info to filter significant commits.
- Enhancing change infos while parsing commits.
  _Also related to:_ #7
- Add wercker badge to README
- [FIX #3] Using wercker as CI.
- [FIX #4] Build changes using commits msg and grouped by version.
- [FIX #1] Create method to parse commit text.
- Add package.json
- Initial commit
"
`;

exports[`changelog creation with tagged commits 1`] = `
"# CHANGELOG

## unpublished
- [FIX #7] Using new info to filter significant commits.

## v0.0.2-beta.2
- Enhancing change infos while parsing commits.
  _Also related to:_ #7

## v0.0.1
- Add wercker badge to README
- [FIX #3] Using wercker as CI.
- [FIX #4] Build changes using commits msg and grouped by version.
- [FIX #1] Create method to parse commit text.
- Add package.json
- Initial commit
"
`;
